notebooks/tutorials/intro_validmind_dataset_model_objects.ipynb

```diff
@@ -0,0 +1,675 @@
+    "    input_id=\"train_dataset\",\n",
+    "    target_column=demo_dataset.target_column,\n",
+    ")\n",
+    "\n",
+    "\n",
+    "model = xgb.XGBClassifier(early_stopping_rounds=10)\n",
+    "model.set_params(\n",
+    "    eval_metric=[\"error\", \"logloss\", \"auc\"],\n",
+    ")\n",
+    "model.fit(\n",
+    "    x_train,\n",
+    "    y_train,\n",
+    "    eval_set=[(x_val, y_val)],\n",
+    "    verbose=False,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, we have `XGBClassifier` model that can be easily wrapped using `init_model` interface of ValidMind."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "vm_model = vm.init_model(\n",
+    "    model=model,\n",
+    "    input_id=\"xgb_model\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Similar to `VMDataset` objcet, we can inspect the methods and attributes of the model now:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "inspect_obj(vm_model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Initialize model evaluation objects and assigning predictions\n",
+    "Now we can store, predictions in the dataset object using `assign_predictions` interface. It adds the prediction column in the dataset to establish the link between model and dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "vm_train_ds.assign_predictions(model=vm_model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The extra prediction column (`xgb_model_prediction`) for the model (`xgb_model`) has been added in the dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(vm_train_ds)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2.5 Custom test using VMDataset and VMModel as inputs\n",
+    "\n",
+    "We will now create a `@vm.metric` wrapper that will allow you to create a reusable metric. Note the following changes in the code below:\n",
+    "\n",
+    "- The function `confusion_matrix` takes two arguments `dataset` and `model`. This is a `VMDataset` and `VMModel` object respectively.\n",
+    "  - `VMDataset` objects allow you to access the dataset's true (target) values by accessing the `.y` attribute.\n",
+    "  - `VMDataset` objects allow you to access the predictions for a given model by accessing the `.y_pred()` method.\n",
+    "- The function docstring provides a description of what the metric does. This will be displayed along with the result in this notebook as well as in the ValidMind platform.\n",
```

Comment:
```suggestion
    "- The function docstring provides a description of what the test does. This will be displayed along with the result in this notebook as well as in the ValidMind platform.\n",
```